static/merge.js

```diff
@@ -400,4 +442,162 @@
 
         remakeTooltips();
     })
+
+    /**
+     * Calculates quantity of observations for each sublevel inside merging class.
+     * @param {Integer} sublevels - Amount of sublevels used in single table.
+     * @param {Integer} constructId - Table's construct ID.
+     * @param {Integer} classId - Merging class ID.
+     */
+    function calculateStarAmountVerticalByLevel(sublevels, constructId, classId) {
+        var star_amount = [];
+
+        for (var i = 0; i < sublevels; i++) {
+            star_amount.push([]);
+        }
+
+        var size = mergedVertical[constructId][classId].length;
+
+        for (var i = 0; i < size; i++) {
+            for (var j = 0; j < sublevels; j++) {
+                const observations = $(mergedVertical[constructId][classId][i].cells[j]).data('modal-launch-observations');
+                
+                if (observations) {
+                    // Add observations and filter reoccurring ones.
+                    observations.map(observation => {
+                        if (!star_amount[j].some(star => observation === star)) {
+                            star_amount[j].push(observation);
+                        }
+                    });
+                }
+
+                $(mergedVertical[constructId][classId][i].student).remove()
+                $(mergedVertical[constructId][classId][i].cells[j]).remove()
+            }
+        }
+
+        return star_amount;
+    }
+
+    /**
+     * Saves students and stars cells for merging class to restore them later
+     * when separate button is clicked.
+     * Cells are saved in this way:
+     * {
+     *   "student": Selector - <th> selector with student name.
+     *   "cells": [] - Array with <td> cells for stars.
+     * }
+     * @param {Array} students - <th> elements with students names.
+     * @param {Integer} constructId - ID of construct.
+     * @param {Integer} classId - ID of merging class. 
+     */
+    function saveCells(students, constructId, classId) {
+        var sublevels = 0;
+
+        if (!mergedVertical.hasOwnProperty(constructId)) {
+            mergedVertical[constructId] = {};
+        }
+
+        mergedVertical[constructId][classId] = []
+
+        for (var i = 0; i < students.length; i++) {
+            var cells = $($(students[i]).parent()).find('td')
+
+            sublevels = cells.length;
+
+            let data = {
+                'student': students[i],
+                'cells': cells
+            };
+
+            mergedVertical[constructId][classId].push(data);
```

Comment:
You can create dict in push method to save a few lines - `data` name has no information in itself